dali/pipeline/executor/executor.cc

```diff
@@ -299,6 +306,24 @@ void Executor<WorkspacePolicy, QueuePolicy>::RunHelper(OpNode &op_node, Workspac
   const auto &schema = spec.GetSchema();
   SmallVector<int, 16> empty_layout_in_idxs;
 
+  cudaStream_t prev_stage_stream = ws.has_stream() && ws.stream() == gpu_op_stream_
+    ? mixed_op_stream_ : gpu_op_stream_;
+
+  for (int i = 0; i < ws.NumOutput(); i++) {
+    auto set_order = [&](auto &output) {
+      // NOTE: the stage streams are synchronized by the executor
+      bool need_sync = output.order().stream() != prev_stage_stream;
+      auto order = ws.has_stream() ? AccessOrder(ws.stream()) : AccessOrder::host();
```

Comment:
I guess you can move this line out of the lambda and do not recompute it over and over.

Reply:
True.